velox/dwio/common/Closeable.h

```diff
@@ -52,7 +52,6 @@ class Closeable {
   virtual void doClose() {}
 
   void destroy() {
-    DCHECK(closed_);
```

Comment:
Why remove this?

Reply:
When underlying logic throws exception, 
1. Closeable::close() is not called
2.  Closeable::~Closeable() would be called, and it does `DCHECK(closed_);` then realize it's not closed yet, then it throws. 

Which is not the right logic, it should first try to close it then check if it's closed, I just moved the check logic after

Reply:
Do we run into this check failure in this PR? Thanks!

Reply:
yes, when testing unsupported compression types in `TableWriterTest`, it won't throw exception anymore but rather fail because of this check, which is not correct behavior